### PR TITLE
new workflow

### PR DESCRIPTION
**Pull request by den-puwanat**
improve the workflow
- has label sync-up
- check verfication steps on trello card
- allow only 1 pull request for 1 card only

WARNING ::  Need to remove adding label when attach pull request on trello card
as new code will add card's label by the conditions

### CODE REVIEW

**Comment by den-puwanat** [comment]
@waf  I might need to remove adding label when attach pull request on trello card 
as new code will add card's label by the conditions

File: index.js [hunk 1]
```diff
@@ -119,61 +139,118 @@ const extractTrelloCardIds = (prBody) =>   {
...
-          }
-        } else {
-          core.info(`trello attachment for card ${cardId} already exists - skipped attachment create.`);
-        }
-      };
-    } else {
-      core.info(`no card urls in pr comment. nothing to do.`);
+
+    if (!cardIds || cardIds.length == 0) {
+      throw new Error("trello-card-check:no card urls in pr comment,please attach your card to this pull request.")
```
Comment (waf) [inline index.js]:
```suggestion
      throw new Error("trello-card-check: no card urls in pr comment, please attach your card to this pull request.")
```

File: index.js [hunk 2]
```diff
@@ -119,61 +139,118 @@ const extractTrelloCardIds = (prBody) =>   {
...
+    }
+
+    if (cardIds.length != 1) {
+      throw new Error("trello-card-check:can not have mulitple trello cards on one pull request.");
```
Comment (waf) [inline index.js]:
```suggestion
      throw new Error("trello-card-check: cannot have multiple trello cards on one pull request.");
```

File: index.js [hunk 3]
```diff
@@ -119,61 +139,118 @@ const extractTrelloCardIds = (prBody) =>   {
...
+    core.info("syncup-label:passed");
+
+
+    // 5. if pull request has [ready for review] label , continue to check if card has verification step provided
+    if (pullrequestHasReviewLabel) {
+      var verificationTexReg = /verification.*step/;
+      var cardObject = await getCard(cardId);
+      var matches = verificationTexReg.exec(cardObject.desc.toLowerCase());
+      if (!matches) {
+        throw Error("verification-step-check:there is no verification steps on card yet , please just put \"Verification Steps\" as text or remove [ready for review] label to skip this error")
```
Comment (waf) [inline index.js]:
```suggestion
        throw Error("verification-step-check: there are no verification steps on the card yet, please put \"Verification Steps\" as text or remove the [ready for review] label to skip this error.")
```

**Review by waf — APPROVED** [review]
Looks good, just minor wording / spelling.

**Comment by den-puwanat** [comment]
Thank you very much :P

**Pull request merged**